src/pal/src/loader/module.cpp

```diff
@@ -1364,6 +1374,37 @@ static LPWSTR LOADGetModuleFileName(MODSTRUCT *module)
     return module->lib_name;
 }
 
+/*
+Function:
+    LOADLoadLibraryDirect [internal]
+
+    Loads a library using a system call, without registering the library with the module list.
+
+Parameters:
+    LPCSTR libraryNameOrPath:           The library to load.
+
+Return value:
+    System handle to the loaded library, or nullptr upon failure (error is set via SetLastError()).
+*/
+static void *LOADLoadLibraryDirect(LPCSTR libraryNameOrPath)
+{
+    _ASSERTE(libraryNameOrPath != nullptr);
+    _ASSERTE(libraryNameOrPath[0] != '\0');
+
+    HMODULE dl_handle = dlopen(libraryNameOrPath, RTLD_LAZY);
```

Comment:
A nit, it seems it would be cleaner to use void* instead of HMODULE here. The result is returned as void* and the dlopen returns void*.